sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,147 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def drift_detect(self) -> Tuple[str, dict]:
+        """
+        Show stack drift for a running stack.
+
+        DETECTION_COMPLETE and DETECTION_FAILED
+        are 2 of the 3 DetectionStatuses from the
+        DescribeStackDriftDetectionStatus API. TIMED_OUT is
+        something we have defined in the event that the wait
+        continues to return DETECTION_IN_PROGRESS after our
+        timeout.
+
+        :returns: The status and resource drifts.
+        """
+        try:
+            status = self._get_status()
+        except StackDoesNotExistError:
+            self.logger.info(f"{self.stack.name} - Does not exist.")
+            status = StackStatus.COMPLETE
+            return (status, {})
+
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        detection_status, stack_drift_status = self._wait_for_drift_status(detection_id)
+
+        return (detection_status, stack_drift_status)
+
+    @add_stack_hooks
+    def drift_show(self) -> Tuple[str, dict]:
+        """
+        Detect drift status on stacks.
+
+        :returns: The status and resource drifts.
+        """
+        try:
+            status = self._get_status()
+        except StackDoesNotExistError:
+            self.logger.info(f"{self.stack.name} - Does not exist.")
+            status = StackStatus.COMPLETE
+            return (status, {})
+
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        detection_status, _ = self._wait_for_drift_status(detection_id)
+
+        if detection_status in ["DETECTION_COMPLETE", "DETECTION_FAILED"]:
+            response = self._describe_stack_resource_drifts()
+        elif detection_status == "TIMED_OUT":
+            response = {}
+        else:
+            raise Exception("Not expected to be reachable")
+
+        return (detection_status, response)
+
+    def _wait_for_drift_status(self, detection_id: str) -> Tuple[str, str]:
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :returns: The drift status.
+        """
+        timeout = 300
+        elapsed = 0
+
+        while True:
+            if elapsed >= timeout:
+                return ("TIMED_OUT", {})
+
+            self.logger.debug(f"{self.stack.name} - Waiting for drift detection")
+            response = self._describe_stack_drift_detection_status(detection_id)
+
+            detection_status = response["DetectionStatus"]
+            stack_drift_status = response["StackDriftStatus"]
+
+            self._log_drift_status(response)
+
+            if detection_status == "DETECTION_IN_PROGRESS":
+                time.sleep(10)
+                elapsed += 10
+            else:
+                return (detection_status, stack_drift_status)
```

Comment:
Since this is a private helper method, I think you should just return the whole response here rather than a tuple of just two bits of info.